gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixTask.java

```diff
@@ -219,6 +228,7 @@ public void cancel() {
     log.info("Gobblin helix task cancellation invoked for jobId {}.", jobId);
     if (this.task != null ) {
       try {
+        this.isCanceled = true;
```

Comment:
set the flag after task.cancel() is done?

Reply:
I set it earlier to make sure when line 171 "this.task.run();" finishes, it can see the flag to be set already. 
As cancel and run are two method runs in different threads, so want to avoid race conditions here. 
If we set it earlier, the worst case is even task finish successfully, as long as we tried to call cancel, we will return as cancel. I think it's acceptable, but let me you WDYT.

Reply:
make sense

Reply:
in some part of gobblin code, we separated these two cases by making two different boolean flags, `cancellationRequested` and `cancelled`

Reply:
Pardon my British english spellings with double L in cancelled :D